mpz-core/src/ggm_tree.rs

```diff
@@ -56,43 +60,102 @@ impl GgmTree {
             }
         }
     }
+
+    /// Reconstruct the GGM tree except the value in a given position.
+    ///
+    /// This reconstructs the GGM tree entirely except `tree[pos] == Block::ZERO`. The bit decomposition of `pos` is the complement of `alpha`. i.e., `pos[i] = 1 xor alpha[i]`.
+    ///
+    /// # Arguments
+    ///
+    /// * `k` - a slice of blocks with length `depth`, the values of k are chosen via OT from k0 and k1. For the i-th value, if alpha[i] == 1, k[i] = k1[i]; else k[i] = k0[i].
+    /// * `alpha` - a slice of bits with length `depth`.
+    /// * `tree` - the destination to write the GGM tree.
+    pub fn reconstruct(&self, tree: &mut [Block], k: &[Block], alpha: &[bool]) {
+        let mut pos = 0;
```

Comment:
do we want some asserts on `tree.len()`, `k.len()`, `alpha.len()` similarly like like in `gen()` above ?

Reply:
yes, we need that asserts.